docs/DeveloperGuide.md

```diff
@@ -2,24 +2,138 @@
+### Add Function
 
+#### System Architecture of the Add Function<br/>
+
+![](team/Architect-digram.PNG)
+
+The Architecture Diagram given above explains the high-level design of the add command.<br/>
+
+Below we will discuss a quick overview of each component.
+
+* UI: The User Interface of the app<br/>
+* Add Command: The main logic command of the add function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+
+#### Design of the Add Function<br/>
+
+UML Class Diagram of the Add function:
+![](team/uml.PNG)<br/>
+
+1) The Add function gets the inputs from the User and places it to the event class so that 
+it can parser to string.<br/>
+2)  It will then pass the event data onto UI class to display and inform user of the event added 
+to the timetable.<br/>
+3) Next,the add function will pass all the parameters (event name,day,time
+and location) to the timetable class into its arraylist divided by the day of the week<br/>
+
+#### Sequence Design of Add Function<br/>
+![](team/seqdiagram.jpg)<br/>
+
+#### Improvements to be made to Version 2.0<br/>
+* To ensure that the users do not enter duplicate event that has the same timing or in between the time 
+that is entered. 
+* A verification check is added to the add command of the version 2.0 of WhereGotTime.
+
+E.g.
+* add /CS2113 Lec /Mon /1200-1400 /LT21<br/>
+* add /CS2113 Lec /Mon /1300-1400 /LT22<br/>
+* add /CS2040C Tut /Mon /1300-1500 /LT32<br/>
+
+The above would result in the first line being add,
+while the other two will result in the app giving an error message
+prompting the user to re-enter again.<br/>
+
+### Edit Function<br/>
+
+### Delete Function<br/>
+
+### Clear Function<br/>
+
+### Display Function<br/>
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](team/CompareCommand_ClassDiagram.jpg) <br/>
+The System Architecture given above explains the high-level design of the <b>COMPARE</b> command.<br/>
+
+Listed below are some features we will be highlighting:<br/>
+
+* Ui: The User Interface of the app<br/>
+* CompareCommand: The main logic command of the <b>COMPARE</b> function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* User: User(s) of the program<br/>
+* UserList: An arraylist of User<br/>
+
+#### Design of the Compare Function<br/>
+![](team/CompareCommand_UML_Diagram.jpg)<br/>
+
+1) The <b>COMPARE</b> function gets the inputs from the User and retrieves the User's / Target User's<br/>
+timetables based on a specific day. It will then compare the timetables and return an arraylist
+of common timeslots.<br/>
+2)  It will then pass the arraylist onto Ui class to display the range of common <br/>
+timeslots the User and Target User have on a specific day of the week.<br/>
+
+
+#### Sequence Design of Compare Function<br/>
+![](team/CompareCommand_SequenceDesign.jpg) <br/>
```

Comment:
![image](https://user-images.githubusercontent.com/53657436/97388356-dfd02d00-1912-11eb-9aba-ccc3d80b50e7.png)

For this diagram it is a bit disorganized and confusing.

Firstly, it is a bit unclear whether the User class here refers to the User class in the compare function (in which case there shouldn't be a random method call coming from it) or if it refers to an actual user (in which case the notation should be the stick figure not the class header)

Secondly, from the CompareCommand activation bar has an unknown return coming from the middle of the activation bar, while the parsedInputs method call being made to the same activation bar is too low down. This is also the case with the activation bar for the self invoked method call execute and the common time array method call to the UI class. 

It is also unclear where the method call to the UI intends to return to as it is chronologically further down than the end of the return from the parsedInputs method.